src/utils/aquarius.ts

```diff
@@ -293,6 +293,7 @@ export async function transformDDOToAssetSelection(
     if (
       priceList[did] &&
       (!didProviderEndpointMap[did] ||
+        datasetProviderEndpoint === undefined ||
```

Comment:
issue, question(non-blocking): shouldn't we check in the same ways as for the other variables with `! datasetProviderEndpoint`, or are there situations where `null` is allowed?

Reply:
I am specifically working with`undefined` here, since it was used in ocean.js and I did not check for `null` interactions